### PR TITLE
shell, dashboard: Edit hosts via the add dialog

### PR DESCRIPTION
**Pull request by mvollmer**
-----

#### Editing hosts in the switcher is more useful

When editing the parameters of a host in the Shell or the Dashboard pages, Cockpit no longer tries to change the name of the remote host.  Instead, it will connect to a different host when you change the address in the "Edit host" dialog.

image

As collateral, Cockpit no longer allows editing of Server Avatars on the Dashboard page. We plan to eventually remove the Dashboard page completely from the Cockpit core. Please contact us if you would like to help with keeping it alive.

### CODE REVIEW

**Comment by marusak** [comment]
So I guess we can close https://github.com/cockpit-project/cockpit/pull/14776 then?

**Comment by mvollmer** [comment]
> So I guess we can close #14776 then?

Yes, if we agree that this one here is the better approach. I think it is.

**Comment by mvollmer** [comment]
`TestHostSwitching.testEdit` is green, so I think we are on the home stretch here. I'll look at the rest of the tests.

**Comment by marusak** [comment]
This will need rebase due to removing of the synchronization of users.

**Comment by mvollmer** [comment]
So I was thinking: should we leave the dashboard alone, and effectively declare it legacy?

**Comment by marusak** [comment]
>  should we leave the dashboard alone, and effectively declare it legacy?

yes. There is plan to drop it from master - and either create a new project in cockpit-project organization with dashboard or just stop supporting it all together.

**Review by marusak — CHANGES_REQUESTED** [review]
For now just tried it. No code review. Here are some notes:

- When I click on the edit button on not the current host, it redirects me to that host instead of opening the dialog. Not sure if that is regression from this PR, I'll compare with master.

- When I edit the host name, it adds a new item. This is unexpected behaviour in my eyes. It should also remove the one, which you are editing. (And if it was your current host then also switch you to the new one).

- I like that when I change user, it just goes through the usual login process!

- Agree with drooping the whole avatar thing

**Comment by marusak** [comment]
> Not sure if that is regression from this PR, I'll compare with master.

seems to happen on master as well. I'll take a look.

**Comment by mvollmer** [comment]
>     * When I click on the edit button on not the current host, it redirects me to that host instead of opening the dialog. Not sure if that is regression from this PR, I'll compare with master.

Yeah, if you click on the pixels of the icon, you get navigation, if you click outside of the icon, you get the edit dialog.

>     * When I edit the host name, it adds a new item. This is unexpected behaviour in my eyes. It should also remove the one, which you are editing. (And if it was your current host then also switch you to the new one).

This is indeed supposed to happen and there is code for it, but I am not sure if we have a test...

**Comment by marusak** [comment]
> Yeah, if you click on the pixels of the icon, you get navigation, if you click outside of the icon, you get the edit dialog.

Fixing it locally. Taking this as a excuse to remove that horrible code where we handle navigation.

**Comment by marusak** [comment]
> Fixing it locally. Taking this as a excuse to remove that horrible code where we handle navigation.

Sent https://github.com/cockpit-project/cockpit/pull/14848

**Comment by mvollmer** [comment]
> > * When I edit the host name, it adds a new item. This is unexpected behaviour in my eyes. It should also remove the one, which you are editing. (And if it was your current host then also switch you to the new one).
> 
> This is indeed supposed to happen and there is code for it, but I am not sure if we have a test...

This still works for me as expected and we have a test for it: https://github.com/cockpit-project/cockpit/pull/14787/files#diff-1925a55cede642a91f2f875a658dc87fe706eb762c68be2807044745822459fdR331

Did I misunderstand what you are doing?

**Comment by marusak** [comment]
> This still works for me as expected and we have a test for it

oh right, it seems to work as well for me. Maybe I had some broken env.

**Comment by mvollmer** [comment]
Let's spin the tests a bit to make sure the failures are still unrelated.

File: pkg/shell/hosts.jsx [hunk 1]
```diff
@@ -10,13 +10,10 @@ import 'polyfills';
 import { superuser } from "superuser";
 import { CockpitNav, CockpitNavItem } from "./nav.jsx";
 
-import { machines } from "machines";
 import { new_machine_dialog_manager } from "machine-dialogs";
 
 import "../../node_modules/@patternfly/patternfly/components/Select/select.scss";
 
-import $ from "jquery"; // Ech... Fixme: We now use the same dialogs as dashboard and we need jQuery to open them
```
Comment (marusak) [inline pkg/shell/hosts.jsx:18]: :heart:

File: pkg/shell/index.html [hunk 2]
```diff
@@ -94,52 +94,6 @@
...
-                                </tr>
-                                <tr id="edit-host-user-row">
-                                    <td class="top" translate="yes">User</td>
-                                    <td>
-                                        <input class="form-control" id="edit-host-user"/>
-                                    </td>
-                                    <td>
-                                        <a tabindex="0" role="button" data-toggle="popover"
-                                            data-trigger="focus" data-placement="bottom" translate="data-content"
-                                            data-content="Leave blank to connect to this machine as the currently logged in user. If you enter a different username, that user will always be used when connecting to this machine.">
```
Comment (marusak) [inline pkg/shell/index.html:121]: This information seems to be missing. Should be present in both add new and edit modals.
Also see https://github.com/cockpit-project/cockpit/issues/14734

**Review by marusak — CHANGES_REQUESTED** [review]
Super nice! Really like that you can relogin with different user, Thank you :) 
One thing I noticed - we lost the info about what happens when you leave the user input empty. Can we get that back? Either as `(i)` popup or just can be mentioned in the text on the top.

File: pkg/shell/index.html (hunk 2 above)
Comment (mvollmer) [reply pkg/shell/index.html:121]: Yes, good point, let's add it back.

Comment (mvollmer) [reply pkg/shell/index.html:121]: In a separate PR. :-)

Comment (marusak) [reply pkg/shell/index.html:121]: ok :) We have issue opened anyway :)

**Review by marusak — APPROVED** [review]
Thanks!

File: pkg/shell/index.html (hunk 2 above)
Comment (mvollmer) [reply pkg/shell/index.html:121]: #14858

**Pull request merged**